lib/handlers/base/api-handler.ts

```diff
@@ -88,22 +88,30 @@ export abstract class APIGLambdaHandler<
   }
 
   get handler(): APIGatewayProxyHandler {
-    return async (event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> => {
-      const handler = this.buildHandler();
+    return metricScope(
+      (metric: MetricsLogger) =>
+        async (event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> => {
+          const requestStart = new Date().getTime();
+          const handler = this.buildHandler();
+          const response = await handler(event, context);
+          const requestEnd = new Date().getTime();
 
-      const response = await handler(event, context);
+          // Track handler duration
+          metric.putDimensions({ [MetricDimension.METHOD]: this.handlerName });
+          metric.putMetric(Metric.HANDLER_DURATION, requestEnd - requestStart, MetricLoggerUnit.Milliseconds);
```

Comment:
Do we need to await this to make sure it gets logged?

Reply:
putMetric doesnt return a promise so no need to await